modules/users/client/services/authentication.client.service.js

```diff
@@ -1,12 +1,81 @@
-'use strict';
+(function() {
+  'use strict';
 
-// Authentication service for user variables
-angular.module('users').factory('Authentication', ['$window',
-  function ($window) {
-    var auth = {
-      user: $window.user
+  angular
+    .module('users')
+    .service('Authentication', Authentication);
+
+  Authentication.$inject = ['$q', 'localStorageService', '$resource', '$http', '$location', '$state'];
+  function Authentication($q, localStorageService, $resource, $http, $location, $state) {
+
+    var readyPromise = $q.defer();
+
+    var service = {
+      ready: readyPromise.promise,
+      user: null,
+      token: null,
+      login: login,
+      signout: signout,
+      refresh: refresh
     };
 
-    return auth;
+    function login(user, token) {
+      setUser(user);
+      setToken(token);
+      setHeader();
+      readyPromise.resolve(service);
+    }
+
+    function setUser(user) {
+      service.user = user;
+    }
+
+    function setToken(token) {
+      service.token = token;
+      localStorageService.set('token', token);
```

Comment:
We could do that, but the token itself has an expiration as well so why not just keep it around in localStorage for as long as its valid?  We could add a check box for keep me logged in that sets the token expiration for like a year and place it in localStorage, otherwise do X(hour or so) amount of time and place it in sessionStorage.  But I am not sure how you would handle provider auths with that method.  Thoughts?